### PR TITLE
Update usage crawlers [sc-6907]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

To optimize the log-parsing crawlers, we introduced the mechanism to do day-by-day log parsing and save the result to metadata history. This PR updates the dataset usage crawlers to add an option of using dataset_usage_history to write the crawler output.

### 🤓 What?

- Add use_history option to crawler config
- Update crawlers to write to dataset_usage_history if configured so, and only fetch 1 day's logs in that case.
- Use UTC start of the day as the measure of datetime

### 🧪 Tested?

- updated unit tests
- tested the three usage crawlers locally

### CODE REVIEW

File: metaphor/common/utils.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from datetime import datetime, time, timedelta, timezone
+
+
+def start_of_day(daysAgo=0) -> datetime:
```
Comment (elic-eon) [inline metaphor/common/utils.py:4]: Is it better to name something else? `start_of_day` sounds like to get the 12 am DateTime of a day to me.

File: metaphor/common/usage_util.py [hunk 2]
```diff
@@ -18,14 +20,29 @@
 class UsageUtil:
     @staticmethod
     def init_dataset(
```
Comment (elic-eon) [inline metaphor/common/usage_util.py:22]: Should we also update tests for `usage_utils` changes?

File: metaphor/common/usage_util.py [hunk 3]
```diff
@@ -99,6 +116,15 @@ def update_table_and_columns_usage(
                     column_name,
                 )
 
+    @staticmethod
+    def update_table_and_columns_usage_history(
```
Comment (elic-eon) [inline metaphor/common/usage_util.py:120]: Also here, to add a new test?

File: metaphor/common/utils.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/common/utils.py:4]: It's exactly `get the 12 am DateTime of a day` in UTC time, by default it's today, but can optionally specify N days ago.

Comment (elic-eon) [reply metaphor/common/utils.py:4]: Oh, my bad. There was a `.date()`

File: metaphor/common/usage_util.py (hunk 3 above)
Comment (alyiwang) [reply metaphor/common/usage_util.py:120]: added some tests in bigquery and redshift usage crawlers

**Review by elic-eon — APPROVED** [review]
LGTM

**Pull request merged**